backend/services/deployments/utils/restutil/options_external_test.go

```diff
@@ -58,130 +54,145 @@ func TestOptionsHandle(t *testing.T) {
-				rest.Get("/path", nil),
-				rest.Options("/path", NewMockOptionsHandler("GET")),
-				rest.Post("/path/path", nil),
-				rest.Options("/path/path", NewMockOptionsHandler("POST")),
-			},
-			[]*rest.Route{
-				rest.Get("/path", nil),
-				rest.Post("/path/path", nil),
+			in: func() *gin.Engine {
+				router := gin.Default()
+				router.GET("/path", dummy)
+				router.POST("/path/path", dummy)
+				return router
+			}(),
+			out: map[string]expHandler{
+				"/path": {
+					http.MethodGet:     dummy,
+					http.MethodOptions: genOptions,
+				},
+				"/path/path": {
+					http.MethodPost:    dummy,
+					http.MethodOptions: genOptions,
+				},
 			},
+			expCount: 4,
 		},
 		{
-			[]*rest.Route{
-				rest.Get("/path", nil),
-				rest.Put("/path", nil),
-				rest.Options("/path", NewMockOptionsHandler("GET", "PUT")),
-				rest.Post("/path/path", nil),
-				rest.Options("/path/path", NewMockOptionsHandler("POST")),
+			in: func() *gin.Engine {
+				router := gin.Default()
+				router.GET("/path", dummy)
+				router.POST("/path/path", dummy)
+				router.PUT("/path", dummy)
+				return router
+			}(),
+			out: map[string]expHandler{
+				"/path": {
+					http.MethodGet:     dummy,
+					http.MethodPut:     dummy,
+					http.MethodOptions: genOptions,
+				},
+				"/path/path": {
+					http.MethodPost:    dummy,
+					http.MethodOptions: genOptions,
+				},
 			},
-			[]*rest.Route{
-				rest.Get("/path", nil),
-				rest.Post("/path/path", nil),
-				rest.Put("/path", nil),
+			expCount: 5,
+		},
+		{
+			in: func() *gin.Engine {
+				router := gin.Default()
+				router.GET("/path", dummy)
+				router.OPTIONS("/path", dummy)
+				return router
+			}(),
+			out: map[string]expHandler{
+				"/path": {
+					http.MethodGet:     dummy,
+					http.MethodOptions: dummy,
+				},
 			},
+			expCount: 2,
 		},
 	}
 
 	for _, test := range testList {
-		out := RouteList(AutogenOptionsRoutes(NewMockOptionsHandler, test.in...))
+		AutogenOptionsRoutes(createHandler, test.in)
 
-		if len(test.out) != len(out) {
+		routes := test.in.Routes()
+		if len(routes) != test.expCount {
 			t.FailNow()
 		}
 
-		// Search requires sorted input
-		sort.Sort(out)
-
-		for _, expectedRoute := range test.out {
-
-			idx := sort.Search(len(out), func(i int) bool {
-				if out[i].HttpMethod+out[i].PathExp >= expectedRoute.HttpMethod+expectedRoute.PathExp {
-					return true
-				}
-				return false
-			})
-
-			if idx == len(out) {
-				t.FailNow()
+		for _, r := range routes {
+			v, ok := test.out[r.Path]
+			if ok != true {
+				t.Errorf("failed with route %+v, route not present", r)
 			}
 
-			if expectedRoute.HttpMethod == "OPTIONS" {
-				exp := &MockResponseWriter{}
-				expectedRoute.Func(exp, nil)
-
-				recived := &MockResponseWriter{}
-				out[idx].Func(recived, nil)
-
-				if !reflect.DeepEqual(exp.methods, recived.methods) {
-					t.FailNow()
-				}
+			h, ok := v[r.Method]
+			if ok != true {
+				t.Errorf("failed with route %+v, method not present", r)
```

Comment:
If the test fails here it will panic on the next condition which might lead to misleading output.
```suggestion
				t.Errorf("failed with route %+v, method not present", r)
				return
```